### ISSUE TITLE
better handling of qphot

### ISSUE THREAD

**Issue by SteveOv** [SteveOv/ebop_maven#71]
One of the params we provide to JKTEBOP task 2 when generating training instances is qphot, the mass ratio used to model mass related stellar distortion.

We don't have any absolute params for the instance, so the current way of calculating this is based on the homology M-R relations applied to k (ratio of radii);
```Python
            # The qphot mass ratio value (MB/MA) affects the lightcurve via the ellipsoidal effect
            # from the distortion of the stars' shape. Generate a value from the ratio of the radii
            # (or set to -1 to force spherical). Standard homology M-R ratios are a starting point;
            # - low mass M-S stars;     M \propto R^2.5
            # - high-mass M-S stars;    M \propto R^1.25
            qphot       = np.random.normal(loc=k**2, scale=0.1)
```

This needs revising to something more sophisticated.  A slight improvement is to randomly choose either the low or high mass relation;
```Python
            qphot       = np.random.normal(loc=np.random.choice([k**2.5, k**1.25], p=[0.75, 0.25]),
                                           scale=0.1)
```
however neither approach models a low mass secondary/high mass primary at all well.

**Comment by SteveOv** [comment]
As an example, with $k=0.2$
$k^{2.5}=0.018$
$k^{1.25}=0.13$

However, if the actual radii were 1.5 and 7.5 $\text{M}_{\odot}$ we would have;
$\frac{1.5^{2.5}}{5^{7.5}}=0.22$

**Comment by SteveOv** [comment]
Some options;
- can I use a relation based on $J$?
- just set qphot to -1 always to force spherical (or perhaps when rA_plus_rB > some threshold)

**Comment by SteveOv** [comment]
Have added qphot to the serialized dataset files (tfrecord) with #72

**Comment by SteveOv** [comment]
If I want to predict qphot I'll need to address the following;
- make qphot fittable in the task3.in.template
  - most flexible approach is to give it a ${qphot_fit} token and default this to 1 in  model_testing.base_jktebop_task3_params()
  - potentially set qphot_fit to 0 for fit_overrides in formal-test-dataset.json (psi Cen, V362 Pav, V530 Ori)
- model_testing
  - fit_against_formal_test_dataset(); add qphot to fit_names
- model_interactive_tester:
  - add qphot to fit_prms (used to read fitted params from jktebop par file)
- the general issue of what to do when there are fit overrides for this
  - the overrides are taken from published works and are used to secure a fit consistent with that published and as such they are essential to the control fit
  - so setting qphot_fit to zero (which is effectively what we have now as this flag is 0 in the template) seems a "no brainer" for the control fit
  - but this isn't always appropriate - when using the estimator, other estimated params are all left free to fit
- did a trial run of predicting qphot (with q = normal(k**2, 1)) and the results were poor
  - the control fits were affected by the qphot label which lead to significant deviation from expected results

**Comment by SteveOv** [comment]
Currently building a training set with qphot fixed at -100

**Comment by SteveOv** [comment]
Some empirical relations fairly widely cited (Demircan & Kahraman, 1991) look to be very useful (esp pp 320);

$k = q^{0.935} \text{ where } M_A \text{ and } M_B < 1.66 \text{M}_{\odot}$

$k = q^{0.542} \text{ where } M_A \text{ and } M_B > 1.66 \text{M}_{\odot}$ 

$k = q^{0.724} \text{ where } M_A > 1.66 \text{M}_{\odot} \text{ and } M_B < 1.66 \text{M}_{\odot}$ (this just doesn't want to render).

These could be directly applicable.

**Comment by SteveOv** [comment]
Having tested models trained of various combinations of the Demircan & Kahraman k-q relations, I've found the best result is to use a relation based on their single R-M relation;
```math
R \simeq 1.01 M^{0.724}, \text{ where } 0.1 \text{M}_{\odot} < M < 18.1 \text{M}_{\odot}
```
giving $q_{phot} \simeq k^{1.4}$. By applying a normal distribution to the exponent the training data has sufficient variation to cover the range of the relations in the previous comment, and gives a Model that performs the best on the test dataset and real systems.
```Python
qphot       = k**np.random.normal(1.4, scale=0.2)
```

**Comment by SteveOv** [comment]
The best of the alternative distributions tried is;

```Python
            if k < 0:
                # Invalid and will cause errors in the qphot calculation
                # This will be discarded as an unusable instance
                qphot = k
            elif k <= 0.8:
                # Secondary significantly smaller, which brings the mixed mass regime into play
                qphot = k**np.random.choice([1.07, 1.85, 1.38], p=[0.50, 0.25, 0.25])
            else:
                qphot = k**np.random.choice([1.07, 1.85], p=[0.50, 0.50])
```

**Comment by SteveOv** [comment]
Did some major testing of various distributions of k, J and qphot with reduced trainsets of 100k instances. The best compromise was to keep the current k & J distribution and revise the qphot distribution to use;
```Python
qphot       = np.random.normal(loc=k**1.4, scale=0.3) if k > 0 else 0
```
yielding the following result for the synthetic test dataset
```
Metrics for 20000 system(s).
----------------------------------------------------------------------------------------------------
Summary    | rA_plus_rB          k          J      ecosw      esinw         bP        MAE        MSE
====================================================================================================
MAE        |   0.007074   0.073505   0.082858   0.023896   0.035243   0.072011   0.049098
MSE        |   0.000115   0.014089   0.068914   0.004474   0.004856   0.013523              0.017662
```
and this for the formal test set (before swapping out V454 Aur for V456 Cyg):
```
Metrics for 23 system(s).
----------------------------------------------------------------------------------------------------
Summary    | rA_plus_rB          k          J      ecosw      esinw         bP        MAE        MSE
====================================================================================================
MAE        |   0.014499   0.112811   0.048683   0.005223   0.026908   0.090902   0.049838
MSE        |   0.000514   0.023675   0.006383   0.000050   0.001427   0.016308              0.008060
``` 
**Note:** these results were produced from a model trained on 80k/20k train/validation instances so the results are not directly comperable to those above which were trained on 200k/50k instance.
